### PR TITLE
libsmartcols: init at v2.36.1

### PR DESCRIPTION
**Pull request by rb2k**
###### Motivation for this change

A requirement for an upcoming package I'm working on (microdnf/libdnf)

This library is a bit weird in that it's part of a different repo (https://github.com/karelzak/util-linux/tree/master/libsmartcols).
That being said, it compiles, I was able to use it as a dependency for libdnf and the build output looks complete (this one s on darwin, hence the dylib):
```
% find /nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/bin
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/include
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/include/libsmartcols
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/include/libsmartcols/libsmartcols.h
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/sbin
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/lib
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/lib/pkgconfig
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/lib/pkgconfig/smartcols.pc
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/lib/libsmartcols.1.dylib
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/lib/libsmartcols.la
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/lib/libsmartcols.dylib
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/man
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/man/man5
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/man/man5/terminal-colors.d.5.gz
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/man/man3
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/man/man8
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/man/man1
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/bash-completion
/nix/store/c0fsi4h9n9yiapwzayx2ac78wvbiqwa7-libsmartcols/share/bash-completion/completions
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rb2k** [comment]
Result of `nixpkgs-review pr 107207` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libsmartcols</li>
  </ul>
</details>

**Comment by rb2k** [comment]
Result of `nixpkgs-review pr 107207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libsmartcols</li>
  </ul>
</details>

File: pkgs/development/libraries/libsmartcols/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  configureFlags = ["--disable-all-programs" "--enable-libsmartcols"];
+
+  buildPhase = ''
+    make libsmartcols.la
+  '';
+
+  installTargets = [ "install-am" "install-pkgconfigDATA"];
+
+  meta = {
+    description = "libsmartcols - pretty output for everyone";
```
Comment (flokli) [inline pkgs/development/libraries/libsmartcols/default.nix:25]: descriptions shouldn't start with the package name - and something like the debian description might be a bit more meaningful:

> smart column output alignment library

File: pkgs/development/libraries/libsmartcols/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 = "0z7nv054pqhlihqiw0vk3h40j0cxk1yxf8zzh0ddmvk6834cnyxs";
+  };
+
+  configureFlags = ["--disable-all-programs" "--enable-libsmartcols"];
+
+  buildPhase = ''
+    make libsmartcols.la
+  '';
+
+  installTargets = [ "install-am" "install-pkgconfigDATA"];
```
Comment (flokli) [inline pkgs/development/libraries/libsmartcols/default.nix:22]:
```suggestion
  installTargets = [ "install-am" "install-pkgconfigDATA" ];
```

This should be consistent with above.

File: pkgs/development/libraries/libsmartcols/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  nativeBuildInputs = [ autoreconfHook pkgconfig python3 ];
+
+  src = fetchFromGitHub {
+    owner = "karelzak";
+    repo = "util-linux";
+    rev = version;
+    sha256 = "0z7nv054pqhlihqiw0vk3h40j0cxk1yxf8zzh0ddmvk6834cnyxs";
+  };
+
+  configureFlags = ["--disable-all-programs" "--enable-libsmartcols"];
```
Comment (flokli) [inline pkgs/development/libraries/libsmartcols/default.nix:16]:
```suggestion
  configureFlags = [ "--disable-all-programs" "--enable-libsmartcols" ];
```

This is how it's formatted with nixpkgs-fmt. Personal preference, feel free to ignore.

**Review by flokli — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libsmartcols/default.nix (hunk 1 above)
Comment (rb2k) [reply pkgs/development/libraries/libsmartcols/default.nix:25]: ah, I took it from  the authors blog post introducing it. But That is an easy change :)

**Comment by rb2k** [comment]
@flokli  Applied the suggestions!

**Pull request merged**